clients/src/test/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImplTest.java

```diff
@@ -1028,9 +1028,9 @@ public void testNewEmptyAssignmentReplacesPreviousOneWaitingOnMetadata() {
 
         verifyReconciliationNotTriggered(membershipManager);
         membershipManager.poll(time.milliseconds());
+        membershipManager.onHeartbeatRequestSent();
```

Comment:
I would suggest we add the check that a reconciliation was triggered here, just adding 
`verifyReconciliationTriggeredAndCompleted(membershipManager, Collections.emptyList());` right after poll. It's part of what this PR is introducing and it completes the pic of what's happening when getting the first (empty) assignment that can be reconciled.

Reply:
Done